src/EhsnPlugin/Plugin.cs

```diff
@@ -11,20 +10,31 @@ public class Plugin : IFieldDataPlugin
     {
         public ParseFileResult ParseFile(Stream fileStream, IFieldDataResultsAppender fieldDataResultsAppender, ILog logger)
         {
-            var eHsn = new Parser()
-                .LoadFromStream(fileStream);
+            try
+            {
+                var parser = new Parser(fieldDataResultsAppender, logger);
+                var eHsn = parser.LoadFromStream(fileStream);
 
-            if (eHsn == null)
-                return ParseFileResult.CannotParse();
+                if (eHsn == null)
+                    return ParseFileResult.CannotParse();
 
-            var targetLocation = fieldDataResultsAppender.GetLocationByIdentifier("MyDummyLocation");
+                try
+                {
+                    parser.Parse(eHsn);
 
-            var now = DateTimeOffset.UtcNow;
-
-            var visit = fieldDataResultsAppender.AddFieldVisit(targetLocation,
-                new FieldVisitDetails(new DateTimeInterval(now.AddHours(-1), now)));
-
-            return ParseFileResult.SuccessfullyParsedAndDataValid();
+                    return ParseFileResult.SuccessfullyParsedAndDataValid();
+                }
+                catch(Exception exception)
+                {
+                    logger.Error($"File can be parsed but an error occurred: {exception.Message}\n{exception.StackTrace}");
+                    return ParseFileResult.SuccessfullyParsedButDataInvalid(exception);
+                }
+            }
+            catch(Exception exception)
+            {
+                logger.Error($"Something went wrong: {exception.Message}\n{exception.StackTrace}");
```

Comment:
This will rarely be hit, because our parser catches all exception and returns null if the uploaded file is non eHSN xml. In that case, plugin returns `CannotParse()` like you said.